src/AssetsPage/AssetsTable/index.jsx

```diff
@@ -1,30 +1,85 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import Table from 'paragon/src/Table';
+import Button from 'paragon/src/Button';
 
-const AssetsTable = ({ assetsList }) => (
-  (!assetsList.length) ? (
-    <span>Loading....</span>
-  ) : (
-    <Table
-      columns={[
-        {
-          label: 'Name',
-          key: 'display_name',
-        },
-        {
-          label: 'Type',
-          key: 'content_type',
-        },
-        {
-          label: 'Date Added',
-          key: 'date_added',
-        },
-      ]}
-      data={assetsList}
-    />
-  )
-);
+import FontAwesomeStyles from 'font-awesome/css/font-awesome.min.css';
+
+class AssetsTable extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      finalAssetsList: [],
+    };
+
+    this.onDeleteClick = this.onDeleteClick.bind(this);
+    this.addDeleteButton = this.addDeleteButton.bind(this);
+  }
+
+  componentDidMount() {
+
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.assetsList.length !== this.props.assetsList.length) {
+      this.addDeleteButton();
+    }
+  }
+
+  onDeleteClick() {
+    this.setState({
+      // TODO: Update application state for Modal and delete logic
+    });
+  }
+
+  addDeleteButton() {
+    const newAssetsList = this.props.assetsList.map((asset) => {
+      const currentAsset = Object.assign({}, asset);
+      const deleteButton = (<Button
+        className={[FontAwesomeStyles.fa, FontAwesomeStyles['fa-trash']]}
+        display={''}
+        buttonType={'light'}
+        aria-label={`Delete current asset named ${currentAsset.display_name}`}
```

Comment:
I might actually opt towards making this a little less verbose (e.g. `Delete asset ${currentAsset.display_name}`), but I'm curious what @cptvitamin's thoughts are.

Reply:
I agree with @arizzitano.  I'm a fan of succinct messaging for everyone.  However, clear, concise instructions are of *paramount* importance for screen reader users who are already exposed to so much verbosity/chatter due to the very nature with which they interact with applications.  I'm not even sure its necessary to include the word "asset" here, since that may already be clear from the context.  Let's go with ```aria-label={`Delete  ${currentAsset.display_name}`}```